src/BCMathExtended/BC.php

```diff
@@ -1,31 +1,28 @@
 <?php
+declare(strict_types=1);
 
 namespace BCMathExtended;
 
-/**
- * Class BC
- * @package BCMathExtended
- */
+use Closure;
+use InvalidArgumentException;
+use function dechex;
+use function hexdec;
+
 class BC
 {
-    const COMPARE_EQUAL = 0;
-    const COMPARE_LEFT_GRATER = 1;
-    const COMPARE_RIGHT_GRATER = -1;
+    public const COMPARE_EQUAL = 0;
+    public const COMPARE_LEFT_GRATER = 1;
+    public const COMPARE_RIGHT_GRATER = -1;
 
-    const DEFAULT_SCALE = 100;
+    protected const DEFAULT_SCALE = 100;
 
-    const MAX_BASE = 256;
+    protected const MAX_BASE = 256;
 
-    const BIT_OPERATOR_AND = 'and';
-    const BIT_OPERATOR_OR = 'or';
-    const BIT_OPERATOR_XOR = 'xor';
+    protected const BIT_OPERATOR_AND = 'and';
+    protected const BIT_OPERATOR_OR = 'or';
+    protected const BIT_OPERATOR_XOR = 'xor';
 
-    /**
-     * @param int|string $number
-     * @param int $precision
-     * @return string
-     */
-    public static function round($number, $precision = 0)
+    public static function round(string $number, int $precision = 0): string
```

Comment:
This will break backwards compatibility, because people pass in any kind of numeric parameter and assume it to work. Also from my understanding `convertScientificNotationToString` is used to convert int|float|string either in normal or scientific notation to a string. Wouldn't it be then useless?

With this strict string requirement for the first parameter, people will need to update their code and do potentially wrong casts to string. The only strict type should be the return type which is always string. I would suggest `round($number, int $precision = 0): string`. And the docblock should have `@param int|float|string $number`.

Reply:
On the other hand if this is a major change, you can do what ever you like :)

Reply:
Yes it's now php7!  with strong types and strict 💪  so It will be released as 5.0